### ISSUE TITLE
examples -> is this correct

### ISSUE THREAD

**Issue by MattiasEppler** [eclipse/paho.mqtt.cpp#182]
I take a look in examples.

async_consume.cpp -> there is a while(true) to consume. This is not async!?
async_subscribe.cpp -> there is no subscribe -> just the callback -> This is async_consume!?

**Comment by fpagliughi** [comment]
Yes. The "consumer" pattern is more common used with the synchronous API. The purpose of the example is to demonstrate that the consumer is actually implemented as part of the async client which underlies the synchronous one. (In this library, the sync client is just a convenience wrapper around async).

A common usage is to have an application with several threads that produce and publish data, and then a single thread that receives and processes all incoming messages (i.e. queue the messages for another thread to process). This can make managing state between messages easier to do within the thread without a lot of locking.

So, in other words,  you can both publish asynchronously and consume synchronously in a single application using a single async client.

**Comment by MattiasEppler** [comment]
Ok. But the name of the examples are not correct. Why is async_subscribe.cpp called subscribe if there is no subscribe inside. This make no sense.

**Comment by frank-metal** [comment]
I'm not sure what you mean. The async_subscribe.cpp example is subscribing to messages via the connected() callback and then receiving messages via the message_arrived() callback.
